features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher-new/source/src/app/components/Apis/Details/Configuration/components/StoreVisibility.jsx

```diff
@@ -38,8 +44,43 @@ import Select from '@material-ui/core/Select';
  * @returns
  */
 export default function StoreVisibility(props) {
+    const [roleValidity, setRoleValidity] = useState(true);
     const { api, configDispatcher } = props;
+    const [invalidRoles, setInvalidRoles] = useState([]);
     const isPublic = api.visibility === 'PUBLIC';
+    const handleRoleAddition = (value, role) => {
+        value.then((resp) => {
+            if (resp) {
+                setRoleValidity(true);
+                configDispatcher({
+                    action: 'visibleRoles',
+                    value: api.visibleRoles.length === 0 ? role : (api.visibleRoles + ',' + role),
+                });
+            } else {
+                setRoleValidity(false);
+                setInvalidRoles([...invalidRoles, role]);
+            }
+        }).catch((error) => {
+            console.error('Error when validating roles ' + error);
+        });
+    };
+
+    const handleRoleDeletion = (role) => {
+        let index = invalidRoles.indexOf(role);
+        const validRoles = [...api.visibleRoles];
```

Comment:
Shall we refactor this code using Array.filter method

i:e:

```javascript
[1,2,3,4].filter(element => element !==3)
```

Reply:
Fixed in f20d89587c00c4aadaf582e15f4f469743872571